### PR TITLE
ETK: Remove old GH actions

### PR DESCRIPTION
**Pull request by noahtallen**
#### Changes proposed in this Pull Request
1. Remove redundant test from GH actions.
2. Refactor TeamCity test order to implicitly include test we were doing on newspack blocks.
3. No longer run any step when the previous fails.

More on the newspack blocks thing:
Several times, the translations for newspack blocks have broken because the textdomain was not fixed correctly before release. Some time ago, I shored that up and added an extra CI check to make sure we didn't break it again. This relied on a separate PHPCS config _just_ to check the textdomain of the synced newspack blocks.

Fast forward to today, newspack blocks now uses a PHPCS standard compliant with ours. Therefore, we can now run our entire lint check against the synced newspack blocks, and it should catch the same issue. This was not possible before because there were other PHPCS issues in the synced blocks.

To accomplish this, we just run the PHP lint check _after_ running the build step. I've split up the build step into two pieces as well to improve flow by adding a "process artifact" step at the very end.

#### Testing instructions

- [x] Builds should continue working
- [x]  After making a change which does not fix newspack blocks textdomain, tests should fail.

### CODE REVIEW

File: .teamcity/_self/builds/WPComPlugins_EditorToolkit.kt [hunk 1]
```diff
@@ -102,36 +96,51 @@ object WPComPlugins_EditorToolKit : BuildType({
 				cd apps/editing-toolkit
 				yarn build
 
-				echo
-				prev_release_build_num=${'$'}(grep build_number ../../etk-release-build/build_meta.txt | sed -e "s/build_number=//")
```
Comment (noahtallen) [inline .teamcity/_self/builds/WPComPlugins_EditorToolkit.kt:106]: This info is already available via the artifact output at the beginning of the job in teamcity

File: .teamcity/_self/builds/WPComPlugins_EditorToolkit.kt [hunk 2]
```diff
@@ -102,36 +95,52 @@ object WPComPlugins_EditorToolKit : BuildType({
...
 					commit_hash=%build.vcs.number%
 					commit_url=https://github.com/Automattic/wp-calypso/commit/%build.vcs.number%
 					build_number=%build.number%
 					EOM
 
+				# 3. Create artifact.
 				echo
 				zip -r ../../../editing-toolkit.zip .
 			"""
-			executionMode = BuildStep.ExecutionMode.RUN_ON_FAILURE
```
Comment (noahtallen) [inline .teamcity/_self/builds/WPComPlugins_EditorToolkit.kt:134]: I don't think we should publish the artifact when the build fails, but I could be convinced either way

**Comment by noahtallen** [comment]
In [the build](https://teamcity.a8c.com/buildConfiguration/calypso_WPComPlugins_EditorToolKit/5978221?showLog=5978221_741_702.1164&logFilter=debug) from commit 1247966, you can see that PHPCS picks up on newspack synced block problems if any managed to get past PHPCBF.

File: .teamcity/_self/builds/WPComPlugins_EditorToolkit.kt [hunk 3]
```diff
@@ -102,36 +95,52 @@ object WPComPlugins_EditorToolKit : BuildType({
...
-				echo "Previous tagged trunk build: ${'$'}prev_release_build_num"
-
 				# Update plugin version in the plugin file and readme.txt.
-				# Note: we also update the previous release build to the same version to restore idempotence
+				# We also update the previous trunk version to match, so that
+				# we can diff the old and new versions without random data.
 				sed -i -e "/^\s\* Version:/c\ * Version: %build.number%" -e "/^define( 'A8C_ETK_PLUGIN_VERSION'/c\define( 'A8C_ETK_PLUGIN_VERSION', '%build.number%' );" ./editing-toolkit-plugin/full-site-editing-plugin.php ../../etk-release-build/full-site-editing-plugin.php
 				sed -i -e "/^Stable tag:\s/c\Stable tag: %build.number%" ./editing-toolkit-plugin/readme.txt ../../etk-release-build/readme.txt
+			"""
+			executionMode = BuildStep.ExecutionMode.RUN_ON_FAILURE
```
Comment (scinos) [inline .teamcity/_self/builds/WPComPlugins_EditorToolkit.kt]: Why do we want to run this script (and the next one) even in if the previous fails?

Comment (noahtallen) [reply .teamcity/_self/builds/WPComPlugins_EditorToolkit.kt]: Good question. I think these were artifacts from the original implementation, so I left them in. I've removed "run_on_failure" from every step in this build.

**Review by fullofcaffeine — APPROVED** [review]
I haven't checked the newspack lint change, but I trust you did :) 

I don't have the experience with the GH actions you removed and teamcity to say with confidence the code changes do what they intend to do, but given you listed "builds continue working" as the acceptance criteria, then we should be good to go 👍🏻

**Pull request merged**